### PR TITLE
Add app name validation APIs

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Add app name validation APIs (ain_validateAppName, /validate_app_name) 

Related issues:
- https://github.com/ainblockchain/ain-blockchain/issues/986

### CODE REVIEW

File: db/index.js [hunk 1]
```diff
@@ -819,6 +820,41 @@ class DB {
...
+    if (!this.isNonExistingApp(appName)) {
+      return {
+        is_valid: false,
+        code: JsonRpcApiResultCode.APP_NAME_ALREADY_IN_USE,
+        message: `App name already in use: ${appName}`
+      };
+    }
+    return {
+      is_valid: true,
+      code: 0
```
Comment (liayoo) [inline db/index.js:854]: JsonRpcApiResultCode.SUCCESS?

**Review by liayoo — APPROVED** [review]
LGTM with a comment. Thanks!

File: db/index.js (hunk 1 above)
Comment (platfowner) [reply db/index.js]: Just done, thanks!

**Comment by platfowner** [comment]
Note that it was amended using isValidServiceName().

**Comment by platfowner** [comment]
Let me merge it now. Thanks for the review. @liayoo

**Pull request merged**